src/Models/Context/OatPageContext/OatPageContext.tsx

```diff
@@ -0,0 +1,610 @@
+                );
+                break;
+            }
+            case OatPageContextActionType.EDIT_PROJECT: {
+                const previousNamespace = draft.currentOntologyNamespace;
+                draft.currentOntologyNamespace = action.payload.namespace.replace(
+                    / /g,
+                    ''
+                );
+                draft.currentOntologyProjectName = action.payload.name;
+                // look through the project data and update any references to the namespace when it changes
+                logDebugConsole(
+                    'debug',
+                    'Updating project namespace to: ' +
+                        draft.currentOntologyNamespace
+                );
+                if (previousNamespace !== draft.currentOntologyNamespace) {
+                    draft.currentOntologyModels.forEach((x) => {
+                        x['@id'] = x['@id'].replace(
+                            `${OAT_MODEL_ID_PREFIX}:${previousNamespace}:`,
+                            `${OAT_MODEL_ID_PREFIX}:${draft.currentOntologyProjectName}:`
+                        );
+                    });
+                }
+                saveData(draft);
+                break;
+            }
+            case OatPageContextActionType.DELETE_PROJECT: {
+                const index = draft.ontologyFiles.findIndex(
+                    (x) => x.id === action.payload.id
+                );
+                if (index >= 0) {
+                    // remove the matching project
+                    draft.ontologyFiles.splice(index, 1);
+
+                    const fileCount = draft.ontologyFiles.length;
+                    if (fileCount > index) {
+                        // switch to the next project in the list
+                        const projectId = draft.ontologyFiles[index].id;
+                        switchCurrentProject(projectId, draft);
+                    } else if (fileCount > 0) {
+                        // switch to the first project in the list
+                        switchCurrentProject(draft.ontologyFiles[0].id, draft);
+                    } else {
+                        // create a new project if none exist to switch to
+                        const name = i18n.t('OATCommon.defaultFileName');
+                        const namespace = OAT_NAMESPACE_DEFAULT_VALUE;
+                        createProject(name, namespace, draft);
+                    }
+                } else {
+                    logDebugConsole(
+                        'warn',
+                        `Could not find project to delete with id: ${action.payload.id}`
+                    );
+                }
+                saveData(draft);
+                break;
+            }
+            case OatPageContextActionType.DUPLICATE_PROJECT: {
+                const id = createGUID();
+                // duplicate the project
+                const project = convertStateToProject(draft);
+                project.projectName =
```

Comment:
why not use ``project.projectName = `${project.projectName}-${i18n.t('OATCommon.duplicateFileNameSuffix')`;`` here?

Reply:
true. why not.

Reply:
Done.